src/commands/Twitch/twitchsubscription.ts

```diff
@@ -41,269 +25,313 @@ type Entry = NotificationsStreamsTwitchStreamer;
 
-		const content = args.t(
-			status === NotificationsStreamsTwitchEventStatus.Offline
-				? LanguageKeys.Commands.Twitch.TwitchSubscriptionAddSuccessOffline
-				: LanguageKeys.Commands.Twitch.TwitchSubscriptionAddSuccessLive,
-			{ name: streamer.display_name, channel: channel.toString() }
+		// Check if there is already a subscription for the given streamer, subscription type, and channel:
+		const alreadyHasEntry = guildSubscriptionsForGuild.some(
+			(guildSubscription) =>
+				guildSubscription.subscription.streamerId === streamer.id && guildSubscription.subscription.subscriptionType === subscriptionType
 		);
-		return send(message, content);
-	}
 
-	public async remove(message: GuildMessage, args: SkyraCommand.Args) {
-		const streamer = await args.pick(UserCommand.streamer);
-		const channel = await args.pick('channelName');
-		const status = await args.pick(UserCommand.status);
-
-		await writeSettings(message.guild, async (settings) => {
-			// then retrieve the index of the entry if the guild already subscribed to them.
-			const subscriptionIndex = settings[this.#kSettingsKey].findIndex((sub) => sub[0] === streamer.id);
+		// If that is the case then throw an error
+		if (alreadyHasEntry) {
+			this.error(LanguageKeys.Commands.Twitch.TwitchSubscriptionAddDuplicated);
+		}
 
-			// If the subscription could not be found, throw.
-			if (subscriptionIndex === -1) this.error(LanguageKeys.Commands.Twitch.TwitchSubscriptionRemoveStreamerNotSubscribed);
+		// Add a new entry to the "guildSubscriptionsForGuild" for streamer, subscription type, channel and message
+		const guildSubscription = new GuildSubscriptionEntity();
+		guildSubscription.guildId = message.guild.id;
+		guildSubscription.channelId = channel.id;
+		guildSubscription.message = customMessage ?? undefined;
 
-			// Retrieve the subscription, then find the index for the notification desired to delete.
-			const subscription = settings[this.#kSettingsKey][subscriptionIndex];
-			const entryIndex = subscription[1].findIndex((entry) => entry.channel === channel.id && entry.status === status);
+		if (streamerForType) {
+			guildSubscription.subscription = streamerForType;
+		} else {
+			// Subscribe to the streamer on the Twitch API, returning the ID of the subscription
+			const subscriptionId = await this.container.client.twitch.subscriptionsStreamHandle(streamer.id, subscriptionType);
+			const twitchSubscriptionEntity = new TwitchSubscriptionEntity();
+
+			// Retrieve the current highest ID in the database, TypeORM doesn't resolve this automatically for some reason
+			const [{ max }] = (await twitchSubscriptions.query(/* sql */ `
+				SELECT max(id)
+				FROM public.twitch_subscriptions;
+			`)) as [MaxQuery];
+
+			// If the max value is not nullish or zero then set it in the entity instance
+			if (!isNullishOrZero(max)) {
+				twitchSubscriptionEntity.id = max + 1;
+			}
```

Comment:
What is this prone-to-error heresy?

Reply:
This is not prone to error. It is literally the same code we use for moderation cases and suggestion ids. In fact it is YOUR code for those parts of Skyra.

Reply:
> It is literally the same code we use for moderation cases and suggestion ids.

No, it's not.

Moderation cases and suggestion IDs increment by one **based** on `guild_id`, which is why it needs to be fetched manually.

Twitch subscriptions are incremented per table, they can use PGSQL's internal counter, which is both atomic and automatic.

We can't use that counter in the other cases because PGSQL doesn't have a per-value counter.